web3/inpage.js

```diff
@@ -174,14 +49,11 @@ if (!window.chrome || !window.chrome.extension) {
 
       // Else, add a full web3 instance
       if (!web3.injectedWeb3) {
-        const rawWeb3 = require('web3/dist/web3.min.js');
-        eval(rawWeb3); // eslint-disable-line no-eval
-
-        const injectedWeb3 = new window.Web3(web3.currentProvider);
-        web3.injectedWeb3 = injectedWeb3;
-        // Expose object directly
-        window.web3 = injectedWeb3;
-        return injectedWeb3[name];
+        require('./lib');
```

Comment:
Okay then ; I just thought it's too bad too include the full Web3 Object for every single page a user will visit whereas it will be useful only for dapps that don't follow the right pattern, as you pointed out.